### PR TITLE
Python log injection

### PR DESCRIPTION
**Pull request by majorgreys**
Documentation for logs injection added to `dd-trace-py` with `v0.20`. This can be merged once this release is completed.

### CODE REVIEW

File: content/tracing/advanced_usage/_index.md [hunk 1]
```diff
@@ -1454,14 +1454,21 @@ If logs are already JSON formatted, there should be nothing left to do.
 [2]: https://docs.datadoghq.com/logs/log_collection/java/?tab=log4j#configure-your-logger
 {{% /tab %}}
 {{% tab "Python" %}}
-Getting the required information needed for logging is easy:
 
-```python
-from ddtrace import helpers
+To inject trace information in Python into logs, you must:
```
Comment (ruthnaebeck) [inline content/tracing/advanced_usage/_index.md]:
```suggestion
To inject trace information from Python into logs:
```

File: content/tracing/advanced_usage/_index.md [hunk 2]
```diff
@@ -1454,14 +1454,21 @@ If logs are already JSON formatted, there should be nothing left to do.
...
+
+1. Set the environment variable `DD_LOGS_INJECTION=true` when using `ddtrace-run` or manually patch the `logging` module.
+2. Update your log formatter to include ``dd.trace_id`` and ``dd.span_id`` attributes from the log record.
+
+The integration with logs occurs as long as the log entry includes:
```
Comment (ruthnaebeck) [inline content/tracing/advanced_usage/_index.md]:
```suggestion
This integration for logs works if the log entry includes:
```

File: content/tracing/advanced_usage/_index.md [hunk 3]
```diff
@@ -1454,12 +1454,37 @@ If logs are already JSON formatted, there should be nothing left to do.
...
-Getting the required information needed for logging is easy:
 
-```python
-from ddtrace import helpers
+To inject trace information from Python into logs:
+
+1. Set the environment variable `DD_LOGS_INJECTION=true` when using `ddtrace-run` or manually patch the `logging` module.
+2. Update your log formatter to include ``dd.trace_id`` and ``dd.span_id`` attributes from the log record.
+
+This integration for logs works if the log entry includes:
```
Comment (brettlangdon) [inline content/tracing/advanced_usage/_index.md]:
```suggestion
This integration for logs works if the log format includes:
```

?

File: content/tracing/advanced_usage/_index.md [hunk 4]
```diff
@@ -1454,12 +1454,37 @@ If logs are already JSON formatted, there should be nothing left to do.
...
+
+- `dd.trace_id=%(dd.trace_id)s`
+- `dd.span_id=%(dd.span_id)s`
+
+An example of this in practice with manual instrumentation:
```
Comment (brettlangdon) [inline content/tracing/advanced_usage/_index.md]: Do we have a `ddtrace-run` example we can put here first?

People will likely just copy/pasta what you have here to try and make it work, we'd like to encourage `ddtrace-run` first, manual patching as the backup.

Comment (majorgreys) [reply content/tracing/advanced_usage/_index.md]: Added `ddtrace-run` example

File: content/tracing/advanced_usage/_index.md [hunk 5]
```diff
@@ -1454,12 +1454,54 @@ If logs are already JSON formatted, there should be nothing left to do.
...
+
+1. Set the environment variable `DD_LOGS_INJECTION=true` when using `ddtrace-run` or manually patch the `logging` module.
+2. Update your log formatter to include ``dd.trace_id`` and ``dd.span_id`` attributes from the log record.
+
+This integration for logs works if the log format includes:
+
+- `dd.trace_id=%(dd.trace_id)s`
+- `dd.span_id=%(dd.span_id)s`
+
+An example of this in practice when using ``ddtrace-run`` with environment variable ``DD_LOGS_INJECTION=true``:
```
Comment (brettlangdon) [inline content/tracing/advanced_usage/_index.md]: We should show the command.


```
DD_LOGS_INJECTION=true ddtrace-run python app.py
```

File: content/tracing/advanced_usage/_index.md [hunk 6]
```diff
@@ -1454,12 +1454,54 @@ If logs are already JSON formatted, there should be nothing left to do.
...
+
+``` python
+import logging
+from ddtrace import tracer
+
+log = logging.getLogger()
```
Comment (brettlangdon) [inline content/tracing/advanced_usage/_index.md]:
```suggestion
log = logging.getLogger(__name__)
```

File: content/tracing/advanced_usage/_index.md [hunk 7]
```diff
@@ -1454,12 +1454,54 @@ If logs are already JSON formatted, there should be nothing left to do.
...
+``` python
+from ddtrace import patch_all; patch_all(logging=True)
+import logging
+from ddtrace import tracer
+
+FORMAT = ('%(asctime)s %(levelname)s [%(name)s] [%(filename)s:%(lineno)d] '
+          '[dd.trace_id=%(dd.trace_id)s dd.span_id=%(dd.span_id)s] '
+          '- %(message)s')
+logging.basicConfig(format=FORMAT)
+log = logging.getLogger()
```
Comment (brettlangdon) [inline content/tracing/advanced_usage/_index.md]:
```suggestion
log = logging.getLogger(__name__)
```

File: content/tracing/advanced_usage/_index.md [hunk 8]
```diff
@@ -1454,12 +1454,36 @@ If logs are already JSON formatted, there should be nothing left to do.
 [2]: https://docs.datadoghq.com/logs/log_collection/java/?tab=log4j#configure-your-logger
 {{% /tab %}}
 {{% tab "Python" %}}
-Getting the required information needed for logging is easy:
 
-```python
-from ddtrace import helpers
+You have two ways to inject trace informations into your logs in Python:
```
Comment (ruthnaebeck) [inline content/tracing/advanced_usage/_index.md]:
```suggestion
You have two ways to inject trace information into your logs in Python:
```

**Pull request merged**